sql: disallow FROM in UPDATE

madelynnblue opened the pull request:
This syntax was parsed but ignored. Remove it because users may expect
it to work since it doesn't produce an error if used.

Fixes #7247

<!-- Reviewable:start -->

---

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/7838)

<!-- Reviewable:end -->

tamird commented:
Is this syntax something we wish to support eventually? I'd feel better about emitting an error lower down.

madelynnblue committed:
sql: disallow FROM in UPDATE

This syntax was parsed but ignored. Produce an error instead of ignoring.

When fixing this issue, make sure to not re-open #7247.

Fixes #7247

mjibson commented:
Agreed. I've marked it as unimplemented and linked an issue.

tamird commented:
LGTM
